yt/frontends/gadget/data_structures.py

```diff
@@ -322,6 +322,7 @@ def __repr__(self):
         return os.path.basename(self.parameter_filename).split(".")[0]
 
     def _get_hvals(self):
+        self.gen_hsmls = False
```

Comment:
maybe log this as debug message

Reply:
I'm not sure why we would want to do that. This is just supposed to be the default value of this attribute in the case that we don't want to generate smoothing lengths (which we would not want to do normally).